### PR TITLE
feat(onKeyStroke): support ignoring repeated stroke

### LINKED ISSUE #2639: onKeyStroke fires rapidly when OS keyrepeat is active
### Clear and concise description of the problem

As a developer using VueUse I want to benefit from keyboard shortcuts / hotkeys so users have an easier time navigating my app. 

unlike vue-hotkey or vue-shortkey back in the Vue 2 world, `onKeyStroke` fires the handler rapidly when keyrepeat is active.

A workaround could be to set `{ eventName: 'keyup' }`, but makes the action feel slow and might not what you expect when you hold down the key. 

Both `keydown` and `keypress` both have the same issue.

I'm sure you could make another workaround using `useDebounceFn`, but I (and many others) always need to fiddle until it works as expected. `useDebounceFn` is not as intuitive. Debouncing is also not a real solution, just another workaround.

### Suggested solution

I suggest to add the function `onKeyStrokeOnce` which works just as `onKeyStroke` but triggers only once on keydown/keypress, then "locks the function", and only unlocks it after `keyup` happened. 

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [x] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by vaakian**
### Description

closes #2639 

Although developer can do the same in the callback, but I think integrating this feature is essential.

### Additional context

reference: [KeyboardEvent.repeat](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/repeat)

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/onKeyStroke/index.ts [hunk 1]
```diff
@@ -105,3 +116,15 @@ export function onKeyPressed(key: KeyFilter, handler: (event: KeyboardEvent) =>
...
+
+/**
+ * Listen for keyboard keys being stroked while ignoring [repeated strokes](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/repeat).
+ *
+ * @see https://vueuse.org/onKeyStroke
+ * @param key
+ * @param handler
+ * @param options
+ */
+export function onKeyStrokeOnce(key: KeyFilter, handler: (event: KeyboardEvent) => void, options: Omit<OnKeyStrokeOptions, 'repeat'> = {}) {
```
Comment (antfu) [inline packages/core/onKeyStroke/index.ts:128]: I think the `Once` is a bit confusing, that may indicate you listen once and dispose. Maybe let's not introduce a new function for that.

File: packages/core/onKeyStroke/index.ts [hunk 2]
```diff
@@ -9,6 +9,12 @@ export interface OnKeyStrokeOptions {
   eventName?: KeyStrokeEventName
   target?: MaybeComputedRef<EventTarget | null | undefined>
   passive?: boolean
+  /**
+   * Set to `false` to ignore repeated strokes.
+   *
+   * @default true
+   */
+  repeat?: boolean
```
Comment (antfu) [inline packages/core/onKeyStroke/index.ts:17]: This is too general, to me, it's more like "we will fire the event" repeatedly.

Instead, I would call it `dedupe` and default to `false`.

Comment (vaakian) [reply packages/core/onKeyStroke/index.ts:17]: I agree with that, but I just found a conventional name `autoRepeat` that seems adopted by public:
1. https://developer.apple.com/documentation/uikit/uistepper/1624079-autorepeat
2. https://learn.microsoft.com/en-us/office/vba/api/access.commandbutton.autorepeat
3. https://doc.qt.io/qt-6/qabstractbutton.html#autoRepeat-prop

File: packages/core/onKeyStroke/index.ts (hunk 1 above)
Comment (vaakian) [reply packages/core/onKeyStroke/index.ts:128]: agree

File: packages/core/onKeyStroke/index.ts [hunk 3]
```diff
@@ -9,6 +10,12 @@ export interface OnKeyStrokeOptions {
   eventName?: KeyStrokeEventName
   target?: MaybeComputedRef<EventTarget | null | undefined>
   passive?: boolean
+  /**
+   * Set to `false` to ignore repeated events when the key is being held down.
+   *
+   * @default true
+   */
+  autoRepeat?: MaybeComputedRef<boolean>
```
Comment (antfu) [inline packages/core/onKeyStroke/index.ts:18]: I think calling it `dedupe` would be more intuitive about what this configs do

**Review by antfu — CHANGES_REQUESTED** [review]

**Comment by katerlouis** [comment]
friendly push? :)

**Review by antfu — APPROVED** [review]

**Pull request merged**